core/internal/filetransfer/tasks.go

```diff
@@ -41,6 +42,9 @@ type Task struct {
 	// Offset is the beginning of the file segment to upload
 	Offset int64
 
+	// Response is the response from the upload or download request.
```

Comment:
Could you document the semantics more? How does it relate to `Err`? Are we guaranteeing that exactly one of (Response, Err) is non-nil? Is the Response always a successful response?